libr/bin/format/mach0/mach0.c

```diff
@@ -34,53 +30,14 @@ typedef struct {
 // USE THIS: int ws = bf->o->info->big_endian;
 #define mach0_endian 1
 
-// TODO: Use the implementation from RUtil
-static ut64 read_uleb128(ulebr *r, ut8 *end) {
-	ut64 result = 0;
-	int bit = 0;
-	ut64 slice = 0;
-	ut8 *p = r->p;
-	do {
-		if (p == end) {
-			eprintf ("malformed uleb128\n");
-			return UT64_MAX;
-		}
-		slice = *p & 0x7f;
-		if (bit > 63) {
-			eprintf ("uleb128 too big for uint64, bit=%d, result=0x%"PFMT64x"\n", bit, result);
-			return UT64_MAX;
-		} else {
-			result |= (slice << bit);
-			bit += 7;
-		}
-	} while (*p++ & 0x80);
-	r->p = p;
-	return result;
+static ut64 read_uleb128(ut8 **p, ut8 *end) {
+	ut64 v;
+	*p = (ut8 *)r_uleb128 (*p, end - *p, &v);
```

Comment:
I think we should check for NULL and when the return value is NULL `read_uleb128` should return `UT64_MAX` to match old behaviour.

Reply:
r_uleb128 from rutil does not return NULL in all scenarios that read_uleb128 did, so we won't cover all cases by just checking for NULL rv

Reply:
maybe we can refactor walk_exports as they did in llvm (https://reviews.llvm.org/rL308690)

Reply:
IMHO `r_uleb128` should be slightly modified to return NULL (aka error) when something goes wrong, like it was done in `read_uleb128`. Right now `r_uleb128` always tries to return something, even if there were errors, but in this way there is no way for the caller of `r_uleb128` to know whether there were problems or not. @trufae do you agree?

Reply:
that's definitely possible, but I have to check that it doesn't break anything. I tried to return with v = UT64_MAX when r_uleb128 and there seems to be a regression in other places.

Reply:
yeah probably some try-and-error will be needed :(

Reply:
yes, but this is probably something to care in a separate PR imho, the behaviour doesnt seems to be different from current code.

Reply:
it is different when running these binaries, [macho-malformed-uleb128.zip](https://github.com/radareorg/radare2/files/5152743/macho-malformed-uleb128.zip). master prints a malformed uleb error while this branch doesn't.